notes/cryptanalysis.tex

```diff
@@ -2,33 +2,200 @@
+
+\subsubsection*{Construction of DES} DES uses a Feistel network construction spanning 16 rounds and a different function at each round. DES at a high level takes an input of 64 bits and permutes it in an initial permutation (ip). Afterward, the 64 bits are split into 32-bit parts, \( L_0, R_0\) which are taken as inputs into the first round of the Feistel network. The input key is 56 bits and DES uses a key schedule that expands the 56-bit key into 16, 48-bit round keys which are used in each round of DES. After all the rounds of the Feistel network, DES runs one final permutation that is the inverse of the initial permutation $ip^{-1}$ before returning the output ciphertext. \newline \newline
+
+
+Let $ip$ be the permutation function and $F : \{0,1\}^{64} \to \{0,1\}^{64}$ the fiestel network function. Then for any message m where (\( |m| = 64 \) bits) and key k where (\( |k| = 56 \) bits), DES can be defined as follows:
+	\[ {E}_{DES} (m,k) = {ip}^{-1}(F({ip}(m), k)) = \cipher \] 
+
+
+ For the given fiestel network function $F$, Let $f_1, f_2, ... f_{16} : \{0,1\}^{32} \to \{0,1\}^{32}$ be the specific round functions of each round of the fiestel permutation and \(k_1, k_2, .... k_16 \) be the round keys used in each round. Then the round functions in DES can be represented as follows: \newline
+\( I = L_0 || R_0 \)  and \(|L_0| = |R_0|\), in other words $L_0$ is the first 32 bits of the initial input I and $R_0$ is the remaining 32 bits  \newline
+\[\forall^{16}_{i=1} i : L_i \leftarrow R_{i-1} \] \[ R_i \leftarrow L_{i-1} \xor {f_i}(R_{i-1}, k_i) \]
+
+
+\begin{figure}%{r}{2in}
+\center
+\begin{tikzpicture}
+   
+    \foreach \pos in {0,...,8} {
+     %\pic at (1.85*\pos,0) {};
+     \node[rectangle, draw] at ($(\pos,0)+(0.1,0.1)$) {S};
+   }
+   
+   \node[rectangle, thick, draw, minimum width = 9cm, minimum height = 1cm] (SBOX) at (4, 0.1) {};
+   
+    \node[circle, draw] (xor) [above of = SBOX, node distance = 1.5cm] {xor};
+
+    \node[rectangle, draw] (pinbox) [above of = SBOX, node distance = 3cm] {Expansion E box};
+    
+    \node[rectangle, draw] (poutbox) [below of = SBOX, node distance = 1.5cm] {mixing Permutation P box};
+
+    \path [draw, ->] (pinbox) -- (xor) {};
+    \path [draw, ->] (xor) -- (SBOX) {};
+    \path [draw, ->] (SBOX) -- (poutbox) {};
+    
+    \node [ below of = poutbox, node distance = 1cm] (out){Output - 32 bits};
+    \node [ above of = pinbox, node distance = 1cm ](in){Input - 32 bits};
+    \path [draw, ->] (in) -- (pinbox)  {};
+    \path [draw, ->] (poutbox) -- (out)  {};
+    
+    \node [circle] (key) [right of = xor, node distance = 5cm] {key-48 bits};
+    \path [draw, ->] (key) -- (xor)  {};
+
+\end{tikzpicture}
+
+\caption{Sbox abstraction}
+\label{fig:sbox}
+\end{figure}
+
+\subsubsection*{DES round functions} Although each round of the fiestel permutation in DES uses a different round function, they follow a similar structure in using the set of auxiliary functions given below: \newline
```

Comment:
HW Question idea: Give a candidate round function for Feistel that only uses linear transforms and have them give an adversary that breaks the cipher.